packages/wallet/src/cip30.ts

```diff
@@ -92,9 +104,33 @@ export const createWalletApi = (
       throw new ApiError(500, 'Nope');
     }
   },
-  getCollateral: async () => {
-    logger.warn('getCollateral is not implemented');
-    return null;
+  getCollateral: async ({ amount = toCbor(MAX_COLLATERAL_AMOUNT) }: { amount?: Cbor } = {}): Promise<Cbor[] | null> => {
+    logger.debug('getting collateral');
+    const wallet = await walletReady;
+    let unspendables = (await firstValueFrom(wallet.utxo.unspendable$)).sort(compareUtxos);
+    if (amount) {
+      try {
+        const filterAmount = CSL.BigNum.from_bytes(Buffer.from(amount, 'hex'));
+        if (filterAmount.compare(MAX_COLLATERAL_AMOUNT) > 0) throw new ApiError(400, 'given amount is too big');
+        const utxos = [];
+        let totalCoins = CSL.BigNum.from_str('0');
+        for (const utxo of unspendables) {
+          const coin = CSL.BigNum.from_str(utxo[1].value.coins.toString());
+          totalCoins = totalCoins.checked_add(coin);
+          utxos.push(utxo);
+          if (totalCoins.compare(filterAmount) !== -1) break;
+        }
+        if (totalCoins.compare(filterAmount) === -1) throw new ApiError(400, 'not enough coins in unspendable UTxOs');
+        unspendables = utxos;
+      } catch (error) {
+        logger.error(error);
+        if (error instanceof ApiError) {
+          throw error;
+        }
+        throw new ApiError(500, 'Nope');
+      }
+    }
+    return Promise.resolve(coreToCsl.utxo(unspendables).map((utxo) => Buffer.from(utxo.to_bytes()).toString('hex')));
```

Comment:
Looks good, just noticed it has a redundant `Promise.resolve` (in `getUtxos` as well). Maybe wrap it in `utxosToCbor` fn to convert the return value of both functions.

```suggestion
    return coreToCsl.utxo(unspendables).map((utxo) => Buffer.from(utxo.to_bytes()).toString('hex'));
```

Reply:
Good catch, I've forgot to change the type of the parameter